packages/language-server/src/xml-view-diagnostics.ts

```diff
@@ -59,9 +56,24 @@ function validationIssuesToLspDiagnostics(
           ...commonDiagnosticPros,
           tags: [DiagnosticTag.Deprecated],
         };
+      case "NoneUniqueIDIssue":
+        return {
+          ...commonDiagnosticPros,
+          relatedInformation: map(currIssue.identicalIDsRanges, (_) => ({
+            // TODO: what message should be used in related issues?
+            message: "also used here",
+            location: {
+              uri: document.uri,
+              range: offsetRangeToLSPRange(_, document),
+            },
+          })),
+        };
       /* istanbul ignore next - defensive programming */
       default:
-        assertNever(issueKind);
+        // We should use assertNever, However, TSC cannot seem to apply exhaustiveness checks
+        // on inner properties (`currentIssue.kind`) nor does it seem to apply type guards correctly
+        // when we first extract the kind (`const issueKind = currIssue.kind)... before the switch.
+        throw Error("None Exhaustive Match");
```

Comment:
As discussed, use `assertNever` here and cast inside the specific `case`

Reply:
done